lib/smack/SmackInstGenerator.cpp

```diff
@@ -882,4 +835,150 @@ void SmackInstGenerator::visitMemSetInst(llvm::MemSetInst& msi) {
+              rep->expr(ii->getArgOperand(0)))));
+    else
+      overApproximateCall(ii);
+  };
+
+  static const auto assignUnFPFuncApp = [&ii, this] (std::string fnBase) {
+    // translation: $res := $<func>.bv*($arg1);
+    if (SmackOptions::FloatEnabled)
+      emit(Stmt::assign(
+            rep->expr(ii),
+            Expr::fn(indexedName(fnBase,
+                {rep->type(ii->getArgOperand(0)->getType())}),
+              rep->expr(ii->getArgOperand(0)))));
+    else
+      overApproximateCall(ii);
+  };
+
+  static const auto assignBinFPFuncApp = [&ii, this] (std::string fnBase) {
+    // translation: $res := $<func>.bv*($arg1, $arg2);
+    if (SmackOptions::FloatEnabled)
+      emit(Stmt::assign(
+            rep->expr(ii),
+            Expr::fn(indexedName(fnBase,
+                {rep->type(ii->getFunctionType()->getReturnType())}),
+              {rep->expr(ii->getArgOperand(0)), rep->expr(ii->getArgOperand(1))})));
+    else
+      overApproximateCall(ii);
+  };
+
+  static const auto assignRoundFPFuncApp = [&ii, this] (std::string rMode) {
+    // translation: $res := $round.bv*(rmode, $arg1);
+    if (SmackOptions::FloatEnabled)
+      emit(Stmt::assign(
+            rep->expr(ii),
+            Expr::fn(indexedName("$round",
+                {rep->type(ii->getFunctionType()->getReturnType())}),
+              {Expr::id(rMode), rep->expr(ii->getArgOperand(0))})));
+    else
+      overApproximateCall(ii);
+  };
+
+  static const auto id = [&ii, this] {
+    // translation: $res := $arg1
+    Value* val = ii->getArgOperand(0);
+    emit(Stmt::assign(rep->expr(ii), rep->expr(val)));
+  };
+
+  static const auto ignore = [&ii, this] {
+    // translation: ignore
+    WARN("ignoring call to "+ii->getCalledFunction()->getName().str());
+    emit(Stmt::skip());
+  };
+
+  // TODO: these functions is consistent with the implementations in math.c,
+  // meaning we can use __builtin_* to implement math.c which is mostly
+  // modeled using __SMACK_code.
+
+  static const std::map<llvm::Intrinsic::ID, std::function<void()>> stmtMap {
+    {llvm::Intrinsic::convert_from_fp16, f16UpCast},
+    {llvm::Intrinsic::convert_to_fp16, f16DownCast},
+    {llvm::Intrinsic::bswap,
+      [] { assignBVFuncApp("$bswap"); }},
+    //TODO: in future versions, there may be llvm.dbg.addr intrinisc
+    {llvm::Intrinsic::dbg_declare, ignore},
+    {llvm::Intrinsic::dbg_value, ignore},
+    {llvm::Intrinsic::expect, id},
+    {llvm::Intrinsic::fabs,
+      [] { assignUnFPFuncApp("$abs"); }},
+    {llvm::Intrinsic::fma,
+      [] { assignUnFPFuncApp("$fma"); }},
+    {llvm::Intrinsic::sqrt,
+      [] { assignUnFPFuncApp("$sqrt"); }},
+    {llvm::Intrinsic::maxnum,
+      [] { assignBinFPFuncApp("$max"); }},
+    {llvm::Intrinsic::minnum,
+      [] { assignBinFPFuncApp("$min"); }},
+    {llvm::Intrinsic::ceil,
+      [] { assignRoundFPFuncApp("RTP"); }},
+    {llvm::Intrinsic::floor,
+      [] { assignRoundFPFuncApp("RTN"); }},
```

Comment:
You should use `RModeLit` here and at other place where rounding mode is hard coded. See:
https://github.com/smackers/smack/blob/develop/include/smack/BoogieAst.h#L90